# Fix mprotect return code handling in _dl_make_stack_executable().

--- glibc-2.3.5/sysdeps/unix/sysv/linux/dl-execstack.c.orig	2005-08-11 14:51:40 +0000
+++ glibc-2.3.5/sysdeps/unix/sysv/linux/dl-execstack.c	2005-08-11 15:20:56 +0000
@@ -89,7 +89,7 @@ _dl_make_stack_executable (void **stack_
 	page -= size;
       else
 	{
-	  if (errno != ENOMEM)	/* Unexpected failure mode.  */
+	  if (errno != ENOMEM && errno != EFAULT)	/* Unexpected failure mode.  */
 	    {
 	      result = errno;
 	      goto out;
@@ -115,7 +115,7 @@ _dl_make_stack_executable (void **stack_
 	page += size;
       else
 	{
-	  if (errno != ENOMEM)	/* Unexpected failure mode.  */
+	  if (errno != ENOMEM && errno != EFAULT)	/* Unexpected failure mode.  */
 	    {
 	      result = errno;
 	      goto out;
